context.go

```diff
@@ -6,53 +6,142 @@ import (
 	"flag"
 	"fmt"
 	"strings"
+	"time"
 )
 
-// Context is a type that is passed through to
-// each Handler action in a cli application. Context
+// Context is an interface that is passed through to
+// each Handler action in a cli application. It
 // can be used to retrieve context-specific args and
 // parsed command-line options.
-type Context struct {
-	context.Context
-	App           *App
-	Command       *Command
+type Context interface {
+	// Functions to be able to manage Context type
+	App() *App
+	SetCommand(command *Command)
+	Command() *Command
+	Context() context.Context
```

Comment:
Do we really need nested contexts?

Reply:
I think the names are unfortunate here, but this isn't really a nested context.

This field refers to the standard library's `context.Context`, so having this function is effectively the same as `http.Request` having a `Context()` function. We're attaching a standard library context to the thing that is already being passed through our application logic, which is our `cli.Context`.

Reply:
I am open to new names. I just extracted the interface out of what we had. As we are going for a v3, new names should not matter much.